File: benchmark/c/cpw/sum_matrix_element_element_integer_division_row_column/sum_matrix_element_element_integer_division_row_column.c


#include <stdio.h>
#include <math.h>
#include <stdlib.h>
#include <limits.h>
#include <stdbool.h>



int sum_matrix_element_element_integer_division_row_column ( int n ) ;
int sum_matrix_element_element_integer_division_row_column ( int n ) {
  int ans = 0;
  for ( int i = 1;
  i <= n;
  i ++ ) for ( int j = 1;
  j <= n;
  j ++ ) ans += ( i / j );
  return ans;
}

File: translation_result/else_bms/fail/c/cpw/sum_matrix_element_element_integer_division_row_column/mistral/Hinted/sum_matrix_element_element_integer_division_row_column.rs


use std::u32;

fn sum_matrix_element_element_integer_division_row_column(n: u32) -> u32 {
let mut ans: u32 = 0;
for i in 1..=n {
for j in 1..=n {
ans = ans.saturating_add(u32::wrapping_div(i, j));
}
}
ans
}

